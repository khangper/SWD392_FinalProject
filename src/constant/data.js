export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
