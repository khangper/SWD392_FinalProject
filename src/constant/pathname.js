export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
