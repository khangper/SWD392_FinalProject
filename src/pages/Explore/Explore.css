.explore-container {
    display: flex;
    justify-content: space-between;
    flex-direction: column;
    padding: 10px;
  }
  .explore-container .search-box {
    display: flex;
    align-items: center;
    padding: 6px 12px;
    border: 2px solid #e9ecef;
    background-color: white;
    border-radius: 2px;
    width: 100%;
    height: 50px;
    margin-bottom: 20px;
  }
  
  .explore-container .search-box input {
    flex: 1;
    border: none;
    outline: none;
    background: transparent;
    font-size: 14px;
    padding: 6px 10px;
  }
  
  .explore-container .search-icon {
    width: 20px;
    height: 20px;
    margin-right: 10px;
  }
  .explore-container .stream-card {
    border-radius: 3px;
    padding: 15px;
    background-position: center;
    background: rgba(51, 1, 51, 0.1);
    text-align: center;
    width: 200px;
    height: 170px;
    flex-shrink: 0;
  }
  .explore-container .featured-courses-section {
    width: 100%;
    padding-top:20px ;
  }
  
  .explore-container .featured-courses {
    display: flex;
    flex-wrap: wrap;
    gap: 20px;
  }
  
  .explore-container .course-card {
    background: #fff;
    border: 1px solid #efefef;
    border-radius: 3px;
    width: 100%;
    max-width: 284px;
    transition: all 0.2s ease-in-out;
  }
  
  .explore-container .course-card:hover {
    box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
  }
  
  .explore-container .course-card img {
    width: 100%;
    border-radius: 3px 3px 0 0;
  }
  
  .explore-container .course-overlay {
    position: absolute;
  top: 10px;
  padding: 10px;
  left: 10px;
  right: 15px;
  bottom: 180px;
  }
  
  .explore-container .badge-seller {
    border-radius: 3px;
  font-size: 10px;
  position: absolute;
  padding: 2px 10px;
  background: #fa8305;
  color: #fff;
  font-family: "Roboto", sans-serif;
  font-weight: 600;
  right: 0;
  top: 10px;
  text-transform: uppercase;
  box-shadow: 0 0 1px 1px rgba(20, 23, 28, 0.1),
    0 3px 1px 0 rgba(20, 23, 28, 0.1);
  }
  
  .explore-container .course-review {
    color: #fff;
  background: #fdcc0d;
  display: inline-block;
  padding: 3px 15px 3px 10px;
  border-radius: 3px;
  font-size: 14px;
  font-weight: 500;
  font-family: "Roboto", sans-serif;
  }
  
  .explore-container .course-review .starIcon {
    width: 15px;
  height: 15px;
  top: 1px;
  position: relative;
  margin-right: 5px;
  }
  
  .explore-container .course-timer {
    bottom: 15px;
    position: absolute;
    right: 0;
    margin-right: 10px;
    color: #fff;
    background: rgba(51, 51, 51, 0.8);
    display: inline-block;
    padding: 3px 10px;
    border-radius: 3px;
    font-size: 12px;
    font-weight: 500;
    font-family: "Roboto", sans-serif;
  }
  
  .explore-container .course-details {
    padding: 10px;
  }
  
  .explore-container .course-details-header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 5px;
  }
  
  .explore-container .course-view-and-date {
    font-size: 12px;
    color: #686f7a;
    margin-bottom: 10px;
  }
  
  .explore-container .course-more-dropdown {
    position: relative;
  }
  
  .explore-container .course-more-dropdown a {
    font-size: 20px;
    color: #afafaf;
    text-decoration: none;
  }
  
  .explore-container .course-more-dropdown:hover .course-more-dropdown-menu {
    display: block;
  }
  
  .explore-container .course-more-dropdown-menu {
    display: none;
    position: absolute;
    right: 0;
    background: #fff;
    border: 1px solid rgba(34, 36, 38, 0.15);
    box-shadow: 0 2px 3px 0 rgba(34, 36, 38, 0.15);
    border-radius: 3px;
    min-width: 160px;
    z-index: 3;
  }
  
  .explore-container .course-more-dropdown-menu span {
    display: block;
    padding: 10px;
    font-size: 14px;
    color: #686f7a;
    cursor: pointer;
  }
  
  .explore-container .course-more-dropdown-menu span:hover {
    background-color: #ffecec;
  }
  
  .explore-container .course-more-dropdown-menu span img {
    width: 14px;
    height: 14px;
    margin-right: 10px;
  }
  
  .explore-container .course-title {
    font-size: 16px;
    font-weight: 600;
    color: #333;
    margin-bottom: 7px;
    display: block;
  }
  
  .explore-container .course-category {
    font-size: 13px;
    color: #686f7a;
    margin-bottom: 15px;
    display: block;
  }
  
  .explore-container .course-info {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
  
  .explore-container .course-author {
    font-size: 14px;
    color: #686f7a;
  }
  
  .explore-container .course-author a {
    color: #333;
    font-weight: 500;
  }
  
  .explore-container .course-price {
    font-size: 18px;
    font-weight: 500;
    color: #333;
  }
  .explore-container .load-more{
    float: left;
    width: 100%;
    margin-bottom: 20px;
    margin-top: 30px;
  }
  .load-more .spinner{
    margin: 0 auto 0;
    width: 50px;
    text-align: center;
    height: 20px;
    border-radius: 100px;
    background-color: #fff;
    line-height: 20px;
    border: 1px solid #efefef;
    cursor: pointer;
  }
  .large-container .explore-container .course-card {
    max-width: 344.9px;
    transition: 0.5s; 
  }
  
  .large-container .explore-container .course-title {
    font-size: 16px; 
  }
  
  .large-container .explore-container .course-category {
    font-size: 14px; 
  }
  
  .large-container .explore-container .course-info {
    align-items: flex-start;
  }
  
  .large-container .explore-container .course-author {
    font-size: 14px; 
  }
  
  .large-container .explore-container .course-price {
    font-size: 18px; 
  }