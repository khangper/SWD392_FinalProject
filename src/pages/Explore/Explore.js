import React, { useEffect, useRef } from "react";
import "./Explore.css";
import search_icon from "..//../assets/search.png";
import ratingStar from "../../assets/rating.png";
import share from "../../assets/share.png";
import saved_course from "../../assets/saved-course.png";
import not_interested from "../../assets/not-interested.png";
import report from "../../assets/report.png";
import { useDispatch, useSelector } from "react-redux";
import { fetchHomeLiveStreamsRequest } from "../../redux/reduxActions/homeActions/HomeLivestreamAction";
import { fetchHomeFeaturedCoursesRequest } from "../../redux/reduxActions/homeActions/HomeFeaturedCourseAction";
import { addSaveCourseRequest } from "../../redux/reduxActions/SaveCourseAction";

const Explore = () => {
  const { liveStreams } = useSelector((state) => state.home_livestream);
  const { featuredCourses } = useSelector((state) => state.home_featuredcourse);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchHomeLiveStreamsRequest());
    dispatch(fetchHomeFeaturedCoursesRequest());
  }, [dispatch]);

  const handleSaveCourse = (course, event) => {
    event.stopPropagation();

    const mappedCourse = {
      id: course.id,
      title: course.title,
      author: course.author || "Unknown Author",
      views: course.views.replace(" Views", "").replace(" views", ""),
      date: course.timeAgo || course.date,
      category: course.category,
      price: course.price,
      hours: course.duration || course.hours,
      rating: course.rating,
      imgSrc: course.imgSrc,
    };

    dispatch(addSaveCourseRequest(mappedCourse));
  };

  const liveStreamRef = useRef(null);
  const scrollLeftLiveStream = () => {
    liveStreamRef.current.scrollBy({ left: -200, behavior: "smooth" });
  };
  const scrollRightLiveStream = () => {
    liveStreamRef.current.scrollBy({ left: 200, behavior: "smooth" });
  };

  const featuredCoursesRef = useRef(null);
  const scrollLeftCourse = () => {
    featuredCoursesRef.current.scrollBy({ left: -410, behavior: "smooth" });
  };
  const scrollRightCourse = () => {
    featuredCoursesRef.current.scrollBy({ left: 410, behavior: "smooth" });
  };

  return (
    <div className="explore-container">
      <section className="search-box">
        <img src={search_icon} alt="Search Icon" className="search-icon" />
        <input
          type="text"
          placeholder="Search for Tuts Videos, Tutors, Tests and more..."
        />
      </section>
      <section className="live-streams-section">
        <div className="live-streams-header">
          <h2 className="live-streams-title">Live Streams</h2>
          <a href="/all_livestream" className="see-all-link">
            See all
          </a>
        </div>
        <div className="live-streams-container">
          <button
            className="scroll-button left"
            onClick={scrollLeftLiveStream}
          ></button>
          <div className="live-streams" ref={liveStreamRef}>
            {liveStreams.map((stream) => (
              <div key={stream.id} className="stream-card">
                <a href="/live_output" className="stream-link">
                  <img src={stream.imgSrc} alt={stream.name} />
                  <h4>{stream.name}</h4>
                  <p>
                    live
                    <span></span>
                  </p>
                </a>
              </div>
            ))}
          </div>
          <button
            className="scroll-button right"
            onClick={scrollRightLiveStream}
          ></button>
        </div>
      </section>

      <section className="featured-courses-section">
        <div className="featured-courses-container">
          <div className="featured-courses">
            {featuredCourses.map((course) => (
              <div key={course.id} className="course-card">
                <a href="/courses-detail-view">
                  <img src={course.imgSrc} alt={course.title} />
                  <div className="course-overlay">
                    <div className="badge-seller">Best seller</div>
                    <div className="course-review">
                      <img className="starIcon" src={ratingStar}></img>
                      {course.rating}
                    </div>
                    <div className="course-timer">{course.hours}</div>
                  </div>
                </a>
                <div className="course-details">
                  <div className="course-details-header">
                    <p className="course-view-and-date">
                      {course.views} views • {course.date}
                    </p>
                    <div className="course-more-dropdown">
                      <a href="#" className="dropdown-button">
                        ⋮
                      </a>
                      <div className="course-more-dropdown-menu">
                        <span>
                          <img src={share} />
                          Share
                        </span>
                        <span onClick={(e) => handleSaveCourse(course, e)}>
                          <img src={saved_course} />
                          Save
                        </span>
                        <span>
                          <img src={not_interested} />
                          Not Interested
                        </span>
                        <span>
                          <img src={report} />
                          Report
                        </span>
                      </div>
                    </div>
                  </div>
                  <a href="#" className="course-title">
                    {course.title}
                  </a>
                  <a href="#" className="course-category">
                    {course.category}
                  </a>
                  <div className="course-info">
                    <p className="course-author">
                      By <a href="#">{course.author}</a>
                    </p>
                    <div className="course-price">{course.price}</div>
                  </div>
                </div>
              </div>
            ))}
          </div>
        </div>
      </section>

      <section className="load-more">
        <div className="spinner">
          <div className="bounce1"></div>
          <div className="bounce2"></div>
          <div className="bounce3"></div>
        </div>
      </section>
    </div>
  );
};

export default Explore;
