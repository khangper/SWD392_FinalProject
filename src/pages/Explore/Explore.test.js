import React from "react";
import { render, screen } from "@testing-library/react";
import Explore from "../Explore/Explore";
import "@testing-library/jest-dom/extend-expect";
import { renderWithProviders } from "../../constant/test-util";

describe("Explore", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  afterAll(() => {
    console.error.mockRestore();
    console.log.mockRestore();
  });
  it("renders the Explore component", async () => {
    const { container } = renderWithProviders(<Explore />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
