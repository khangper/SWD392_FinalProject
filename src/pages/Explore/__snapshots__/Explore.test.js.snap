// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explore renders the Explore component 1`] = `
<div>
  <div
    class="explore-container"
  >
    <section
      class="search-box"
    >
      <img
        alt="Search Icon"
        class="search-icon"
        src="search.png"
      />
      <input
        placeholder="Search for Tuts Videos, Tutors, Tests and more..."
        type="text"
        value=""
      />
    </section>
    <section
      class="live-streams-section"
    >
      <div
        class="live-streams-header"
      >
        <h2
          class="live-streams-title"
        >
          Live Streams
        </h2>
        <a
          class="see-all-link"
          href="/all_livestream"
        >
          See all
        </a>
      </div>
      <div
        class="live-streams-container"
      >
        <button
          class="scroll-button left"
        />
        <div
          class="live-streams"
        />
        <button
          class="scroll-button right"
        />
      </div>
    </section>
    <section
      class="featured-courses-section"
    >
      <div
        class="featured-courses-container"
      >
        <div
          class="featured-courses"
        />
      </div>
    </section>
  </div>
</div>
`;

exports[`Home renders the Home component 1`] = `
<div>
  <div
    class="explore-container"
  >
    <section
      class="search-box"
    >
      <img
        alt="Search Icon"
        class="search-icon"
        src="search.png"
      />
      <input
        placeholder="Search for Tuts Videos, Tutors, Tests and more..."
        type="text"
        value=""
      />
    </section>
    <section
      class="live-streams-section"
    >
      <div
        class="live-streams-header"
      >
        <h2
          class="live-streams-title"
        >
          Live Streams
        </h2>
        <a
          class="see-all-link"
          href="/all_livestream"
        >
          See all
        </a>
      </div>
      <div
        class="live-streams-container"
      >
        <button
          class="scroll-button left"
        />
        <div
          class="live-streams"
        />
        <button
          class="scroll-button right"
        />
      </div>
    </section>
    <section
      class="featured-courses-section"
    >
      <div
        class="featured-courses-container"
      >
        <div
          class="featured-courses"
        />
      </div>
    </section>
  </div>
</div>
`;
